internal/engine/session_integration_test.go

```diff
@@ -651,3 +665,14 @@ func TestNewOrchestraClientProbeServicesNewClientFailure(t *testing.T) {
 		t.Fatal("expected nil client here")
 	}
 }
+
+func TestSessionNewSubmitterWorks(t *testing.T) {
+	sess := newSessionForTesting(t)
+	subm, err := sess.NewSubmitter(context.Background())
+	if err != nil {
+		t.Fatal(err)
+	}
+	if subm == nil {
+		t.Fatal("expected non nil submitter here")
+	}
```

Comment:
This test is rather minimal because we test also elsewhere. The main reason to have it is to make sure we respect the basic API.